### PR TITLE
PR: Add astropy and pythorch to UMR blacklist by default.

### LINKED ISSUE #6962: Add Astropy to default UMR blacklist for Python 2.7 due to resulting bug
## Problem Description

Need a place to document known Spyder issue with Astropy. There is a PR at astropy/astropy#7374 but we feel it belongs more to your documentation over here. Please advise.

### What steps will reproduce the problem?

Detailed description at astropy/astropy#7363

### What is the expected output? What do you see instead?

Expect no error message. Instead, there is error message.

## Package Versions

* Spyder: 2.3.5.2 - 3.1.2
* Python: 2.7
* Qt: N/A
* PyQt: N/A 
* Operating System: openSUSE Leap 42.3

### Dependencies

N/A

### LINKED ISSUE #7041: Multiple loading of pytorch module with script causes RuntimeError on IPython console.
## Problem Description
Importing pytorch more than twice with running a file on the editor window causes a runtime error in the IPython console.
I'm using pytorch 0.4.0.

### What steps will reproduce the problem?

1. Run Spyder3.
2. Write 'import torch' in the editor window.
3. Run the script.
4. Run the script again.

### What is the expected output? What do you see instead?

Expecting no exception. But some compiling/reloading existing module seems to cause an error.
Note that steps below are the dirty weird solution.
1. Run Spyder3.
2. Write 'import torch' in the **IPython console** and press enter.
3. Write 'import torch' in the editor window.
4. Run the script.
5. Run the script again.
In short, **initially loading pytorch module on IPython console solves the problem**, but I don't know why.

### Paste Traceback/Error Below (if applicable)

```python-traceback
Traceback (most recent call last):

  File "<ipython-input-3-bc7ab1fdfb9b>", line 1, in <module>
    runfile('/Users/aaa/.spyder-py3/temp.py', wdir='/Users/aaa/.spyder-py3')

  File "/usr/local/lib/python3.6/site-packages/spyder/utils/site/sitecustomize.py", line 705, in runfile
    execfile(filename, namespace)

  File "/usr/local/lib/python3.6/site-packages/spyder/utils/site/sitecustomize.py", line 102, in execfile
    exec(compile(f.read(), filename, 'exec'), namespace)

  File "/Users/aaa/.spyder-py3/temp.py", line 7, in <module>
    import torch

  File "/usr/local/lib/python3.6/site-packages/torch/__init__.py", line 184, in <module>
    from .tensor import Tensor

  File "/usr/local/lib/python3.6/site-packages/torch/tensor.py", line 12, in <module>
    class Tensor(torch._C._TensorBase):

  File "/usr/local/lib/python3.6/site-packages/torch/tensor.py", line 170, in Tensor
    """)

RuntimeError: method 'detach' already has a docstring

```

## Package Versions

* Spyder: 3.2.8
* Python: 3.6.4
* Qt:
* PyQt:
* Operating System: macOS 10.13.4

### Dependencies

```
IPython >=4.0    :  6.3.1 (OK)
cython >=0.21    :  0.28.2 (OK)
jedi >=0.9.0     :  0.12.0 (OK)
nbconvert >=4.0  :  5.3.1 (OK)
numpy >=1.7      :  1.14.1 (OK)
pandas >=0.13.1  :  0.22.0 (OK)
psutil >=0.3     :  5.4.5 (OK)
pycodestyle >=2.3:  2.4.0 (OK)
pyflakes >=0.6.0 :  1.6.0 (OK)
pygments >=2.0   :  2.2.0 (OK)
pylint >=0.25    :  1.8.4 (OK)
qtconsole >=4.2.0:  4.3.1 (OK)
rope >=0.9.4     :  0.10.7 (OK)
sphinx >=0.6.6   :  1.7.4 (OK)
sympy >=0.7.3    :  1.1.1 (OK)
```

### PR DESCRIPTION
**Pull request by dalthviz**
### Pull Request Checklist

* [x] Read and followed this repo's [Contributing Guidelines](https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md)
* [x] Based your PR on the latest version of the correct branch (master or 3.x)
* [x] Followed [PEP8](https://www.python.org/dev/peps/pep-0008/) for code style
* [x] Ensured your pull request hasn't eliminated unrelated blank lines/spaces,
      modified the ``spyder/defaults`` directory, or added new icons/assets
* [x] Wrote at least one-line docstrings for any new functions
* [x] Added at least one unit test covering the changes, if at all possible
* [x] Described your changes and the motivation for them below
* [x] Noted what issue(s) this pull request resolves, creating one if needed
* [x] Included a screenshot, if this PR makes any visible changes to the UI

## Description of Changes

Add `astropy` and `pytorch` by default to the UMR blacklist

### Issue(s) Resolved

Fixes #7041 
Fixes #6962

### CODE REVIEW

File: spyder/utils/site/sitecustomize.py [hunk 1]
```diff
@@ -555,7 +555,10 @@ def __init__(self, namelist=None, pathlist=None):
             namelist = []
         spy_modules = ['sitecustomize', 'spyder', 'spyderplugins']
         mpl_modules = ['matplotlib', 'tkinter', 'Tkinter']
-        self.namelist = namelist + spy_modules + mpl_modules
+        other_modules = ['pytorch']
+        if PY2:
+            other_modules.append('astropy')
```
Comment (ccordoba12) [inline spyder/utils/site/sitecustomize.py]: Please add comment with a reference to the issues that lead us to blacklist each of these modules, one comment per module.

Thanks!

File: spyder/utils/site/sitecustomize.py [hunk 2]
```diff
@@ -555,7 +555,13 @@ def __init__(self, namelist=None, pathlist=None):
             namelist = []
         spy_modules = ['sitecustomize', 'spyder', 'spyderplugins']
         mpl_modules = ['matplotlib', 'tkinter', 'Tkinter']
-        self.namelist = namelist + spy_modules + mpl_modules
+        # Needed to add modules to the UMR blacklist.
```
Comment (ccordoba12) [inline spyder/utils/site/sitecustomize.py]: Please rephrase this like

    # Add other, necessary modules to the blacklist

Comment (ccordoba12) [reply spyder/utils/site/sitecustomize.py]: Sorry, this is better:

    # Add other, necessary modules to the UMR blacklist

**Comment by pllim** [comment]
c/c @ dhomeier 

Thank you, everyone!

**Comment by dhomeier** [comment]
Thanks as well! FI the `fastmat` project had reported the [same issue](https://github.com/EMS-TU-Ilmenau/fastmat/issues/27) under `PY2` and might best be added to the list as well.

**Comment by ccordoba12** [comment]
@dhomeier, could you open a new issue for `fastmat`? Thanks!

**Comment by ccordoba12** [comment]
@dhomeier, no worries, I opened an issue already.

**Comment by dhomeier** [comment]
Thanks, @ccordoba12! I am actually not familiar with `fastmat`, it was just a related case we ran into when looking for reports of similar problems. Looks like this should work, but @JanKirchhofTU or @ChristophWWagner could probably confirm better.

**Pull request merged**